### ISSUE TITLE
4.6.4 Causing Kernel BUG in Truenas Scale Cobia

### ISSUE THREAD

**Issue by kuantek** [Trigus42/alpine-qbittorrentvpn#63]
I am running a NAS on Truenas Scale Cobia.

Qbittorrent downloads to a local SSD, then moves the completed download to the NAS. I have no idea how this is possible, but the container (just starting the container, not downloading anything) was causing a memory overwrite bug on the NAS, causing it to boot loop.

I found that the qbittorrent container was causing this by trial and error and elimination of other possibilities. Upon downgrading the container to 4.6.3, the issue has gone away.

**Comment by Trigus42** [comment]
Do you have any more info? I have not changed anything since `4.6.3` and "memory overwrite bug" is a pretty broad term. What lead you to believe there is a some sort of memory corruption going on? Also did the NAS boot loop or just the container?

**Comment by kuantek** [comment]
Yeah, I should probably cancel this. It was happenstance that downgrading the container did anything at all.

tldr; Switching from SMB to NFS mounting on the docker host has completely solved this issue.

It ended up being a bug with Truenas and SMB, but I haven't seen anything about it and I don't know what else to do.

For some reason, SMB was no longer showing the drive space, and was being weird about moving stuff (the only thing I saw in the samba log was the durable handles were failing to close, I truly have no idea what was going on and I was pulling my hair out all day yesterday trying to fix everything).

Basically, when Sabnzbd or qbittorrent finished a download and attempted to move the file from the local download folder to the NAS SMB complete folder, it wouldn't appear to work (the file would end up in the correct place in the end, somehow) but it was causing the memory overwrite and kernel panic, so the NAS would restart. Upon restart, qbitorrent/sabnzbd wouldn't know that the file had been copied correctly and would try again, even though the file WAS on the share. So it would just keep on moving files, the file move would fail, and truenas would kernel panic and reset. Ad nauseum.
